darshan-util/pydarshan/tests/test_summary.py

```diff
@@ -122,3 +125,229 @@ def test_main_all_logs_repo_files(tmpdir, log_repo_files):
+
+        # get the time stamps in the current time zone
+        expected_start = str(datetime.fromtimestamp(R.report.metadata["job"]["start_time"]))
+        expected_end = str(datetime.fromtimestamp(R.report.metadata["job"]["end_time"]))
+        assert actual_metadata_df.loc["Start Time"].values[0] == expected_start
+        assert actual_metadata_df.loc["End Time"].values[0] == expected_end
+
+
+    @pytest.mark.parametrize(
+        "log_path",
+        [
+            # each of these logs offers a unique
+            # set of modules to verify
+            "tests/input/sample.darshan",
+            "tests/input/noposix.darshan",
+            "tests/input/noposixopens.darshan",
+            "tests/input/sample-goodost.darshan",
+            "tests/input/sample-dxt-simple.darshan",
+        ],
+    )
+    def test_module_table(self, log_path):
+        # regression test for `summary.ReportData.get_module_table()`
+
+        # collect the report data
+        R = summary.ReportData(log_path=log_path)
+        # convert the module table back to a pandas dataframe
+        actual_mod_df = pd.read_html(R.module_table, index_col=0)[0]
+
+        # verify the number of modules in the report is equal to
+        # the number of rows in the module table
+        expected_module_count = len(R.report.modules.keys())
+        assert expected_module_count == actual_mod_df.shape[0]
+
+        # check each entry of the module dataframe by retrieving all row
+        # labels and checking they match their values
+        if "sample.darshan" in log_path:
+            assert actual_mod_df.loc["POSIX (ver=3)"].values[0] == "0.18 KiB"
+            assert actual_mod_df.loc["MPI-IO (ver=2)"].values[0] == "0.15 KiB"
+            assert actual_mod_df.loc["LUSTRE (ver=1)"].values[0] == "0.08 KiB"
+            assert actual_mod_df.loc["STDIO (ver=1)"].values[0] == "3.16 KiB"
+        elif "noposix.darshan" in log_path:
+            assert actual_mod_df.loc["LUSTRE (ver=1)"].values[0] == "6.07 KiB"
+            assert actual_mod_df.loc["STDIO (ver=1)"].values[0] == "0.21 KiB"
+        elif "noposixopens.darshan" in log_path:
+            assert actual_mod_df.loc["POSIX (ver=3)"].values[0] == "0.04 KiB"
+            assert actual_mod_df.loc["STDIO (ver=1)"].values[0] == "0.27 KiB"
+        elif "sample-goodost.darshan" in log_path:
+            assert actual_mod_df.loc["POSIX (ver=3)"].values[0] == "5.59 KiB"
+            assert actual_mod_df.loc["LUSTRE (ver=1)"].values[0] == "1.47 KiB"
+            assert actual_mod_df.loc["STDIO (ver=1)"].values[0] == "0.07 KiB"
+        elif "sample-dxt-simple.darshan" in log_path:
+            assert actual_mod_df.loc["POSIX (ver=4)"].values[0] == "2.94 KiB"
+            assert actual_mod_df.loc["MPI-IO (ver=3)"].values[0] == "1.02 KiB"
+            assert actual_mod_df.loc["DXT_POSIX (ver=1)"].values[0] == "0.08 KiB"
+            assert actual_mod_df.loc["DXT_MPIIO (ver=2)"].values[0] == "0.06 KiB"
```

Comment:
For situations like this when you are taking a dataframe and manually decomposing it into individual rows and values, that's often a hint that an array-based approach might be better. Two other problems here include: 1) the parametrized test has to awkwardly check the name of the input parameter every time; 2) when you get a failure, you don't necessarily know if all fields fail instead of just 1 (which NumPy/pandas testing matching should tell you with an i.e., 1/4 mismatch indicator, etc.).

I think there may be a few tests here that might benefit from such an adjustment, but for this particular one I did the work already, and something more concise/efficient may be available from the pandas series/DF test/assertion functions (please do take a look).

```diff
--- a/darshan-util/pydarshan/tests/test_summary.py
+++ b/darshan-util/pydarshan/tests/test_summary.py
@@ -5,6 +5,7 @@ from unittest import mock
 from datetime import datetime
 
 import pandas as pd
+from numpy.testing import assert_array_equal
 
 import darshan
 from darshan.cli import summary
@@ -245,18 +246,28 @@ class TestReportData:
 
 
     @pytest.mark.parametrize(
-        "log_path",
+        "log_path, exp_ind, exp_vals",
         [
             # each of these logs offers a unique
             # set of modules to verify
-            "tests/input/sample.darshan",
-            "tests/input/noposix.darshan",
-            "tests/input/noposixopens.darshan",
-            "tests/input/sample-goodost.darshan",
-            "tests/input/sample-dxt-simple.darshan",
+            ("tests/input/sample.darshan",
+            ['POSIX (ver=3)', 'MPI-IO (ver=2)', 'LUSTRE (ver=1)', 'STDIO (ver=1)'],
+            [["0.18 KiB"], ["0.15 KiB"], ["0.08 KiB"], ["3.16 KiB"]]),
+            ("tests/input/noposix.darshan",
+            ['LUSTRE (ver=1)', 'STDIO (ver=1)'],
+            [["6.07 KiB"], ["0.21 KiB"]]),
+            ("tests/input/noposixopens.darshan",
+            ['POSIX (ver=3)', 'STDIO (ver=1)'],
+            [["0.04 KiB"], ["0.27 KiB"]]),
+            ("tests/input/sample-goodost.darshan",
+            ['POSIX (ver=3)', 'LUSTRE (ver=1)', 'STDIO (ver=1)'],
+            [["5.59 KiB"], ["1.47 KiB"], ["0.07 KiB"]]),
+            ("tests/input/sample-dxt-simple.darshan",
+            ['POSIX (ver=4)', 'MPI-IO (ver=3)', 'DXT_POSIX (ver=1)', 'DXT_MPIIO (ver=2)'],
+            [["2.94 KiB"], ["1.02 KiB"], ["0.08 KiB"], ["0.06 KiB"]])
         ],
     )
-    def test_module_table(self, log_path):
+    def test_module_table(self, log_path, exp_ind, exp_vals):
         # regression test for `summary.ReportData.get_module_table()`
 
         # collect the report data
@@ -269,28 +280,9 @@ class TestReportData:
         expected_module_count = len(R.report.modules.keys())
         assert expected_module_count == actual_mod_df.shape[0]
 
-        # check each entry of the module dataframe by retrieving all row
-        # labels and checking they match their values
-        if "sample.darshan" in log_path:
-            assert actual_mod_df.loc["POSIX (ver=3)"].values[0] == "0.18 KiB"
-            assert actual_mod_df.loc["MPI-IO (ver=2)"].values[0] == "0.15 KiB"
-            assert actual_mod_df.loc["LUSTRE (ver=1)"].values[0] == "0.08 KiB"
-            assert actual_mod_df.loc["STDIO (ver=1)"].values[0] == "3.16 KiB"
-        elif "noposix.darshan" in log_path:
-            assert actual_mod_df.loc["LUSTRE (ver=1)"].values[0] == "6.07 KiB"
-            assert actual_mod_df.loc["STDIO (ver=1)"].values[0] == "0.21 KiB"
-        elif "noposixopens.darshan" in log_path:
-            assert actual_mod_df.loc["POSIX (ver=3)"].values[0] == "0.04 KiB"
-            assert actual_mod_df.loc["STDIO (ver=1)"].values[0] == "0.27 KiB"
-        elif "sample-goodost.darshan" in log_path:
-            assert actual_mod_df.loc["POSIX (ver=3)"].values[0] == "5.59 KiB"
-            assert actual_mod_df.loc["LUSTRE (ver=1)"].values[0] == "1.47 KiB"
-            assert actual_mod_df.loc["STDIO (ver=1)"].values[0] == "0.07 KiB"
-        elif "sample-dxt-simple.darshan" in log_path:
-            assert actual_mod_df.loc["POSIX (ver=4)"].values[0] == "2.94 KiB"
-            assert actual_mod_df.loc["MPI-IO (ver=3)"].values[0] == "1.02 KiB"
-            assert actual_mod_df.loc["DXT_POSIX (ver=1)"].values[0] == "0.08 KiB"
-            assert actual_mod_df.loc["DXT_MPIIO (ver=2)"].values[0] == "0.06 KiB"
+        # check the module dataframes
+        assert_array_equal(actual_mod_df.index, exp_ind)
+        assert_array_equal(actual_mod_df.values, exp_vals)

```